### PR TITLE
add fix for notification role, also fix typing for install task

### LINKED ISSUE #7: [BUG] missing notification service-role
**Describe the bug**
Module takes most settings for notifications, but misses the role for it

**To Reproduce**
1. Define notification settings
2. Go to ui, edit the terraform created task
3. note that notification arn is there, but no role
4. create a role and add it to the task by hand, so we can do `terraform plan` and see where it should be put [1]

**Expected behavior**
take input for notification role

[1]
**Screenshots**
```
 # module.patch_manager.aws_ssm_maintenance_window_task.scan will be updated in-place
  ~ resource "aws_ssm_maintenance_window_task" "scan" {
        id               = "someUuid"
        # (7 unchanged attributes hidden)


      ~ task_invocation_parameters {

          ~ run_command_parameters {
              - service_role_arn     = "arn:aws:iam::<accountId>:role/test-role" -> null
                # (3 unchanged attributes hidden)


                # (2 unchanged blocks hidden)
            }
        }
        # (1 unchanged block hidden)
    }
```

**Desktop (please complete the following information):**

**Additional context**
Related issue in provider: https://github.com/hashicorp/terraform-provider-aws/issues/17624

### PR DESCRIPTION
**Pull request by davidkarlsen**
Fixes #7

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -191,14 +197,22 @@ variable "install_log_prefix" {
...
   description = "A set of objects containing `notification_config`s [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_maintenance_window_task#notification_config)"
 
   type = set(object({
-    notification_arn    = string
-    notification_events = string
-    notification_type   = string
+    notification_arn      = string
+    notification_events   = list(string)
+    notification_type     = string
+    notification_role_arn = string
```
Comment (sblack4) [inline variables.tf:215]:
```suggestion
```

This looks like an oopsie

**Review by sblack4 — CHANGES_REQUESTED** [review]
Beautiful, there's just one thing that I think got accidentally added. Or I might be confuse. ❤️

**Comment by smiller171** [comment]
@davidkarlsen Is there a good reason not to create the role within the module?

File: variables.tf (hunk 1 above)
Comment (davidkarlsen) [reply variables.tf]: An oopsie indeed :-)

**Comment by davidkarlsen** [comment]
> @davidkarlsen Is there a good reason not to create the role within the module?

that's maybe a tad busy. there are probably things to be done outside of the module anyways, like key-grants etc.

**Review by sblack4 — APPROVED** [review]
This looks good to me. @smiller171 IDK why we would want to try and manage IAM roles in the module. IMO it should do one thing and do it well

**Comment by sblack4** [comment]
I'ma get this merge in, we can add IAM stuff later if need be

**Comment by smiller171** [comment]
Yeah, this is definitely the minimal change to fix the bug.

**Pull request merged**